app/product_notices.go

```diff
@@ -33,7 +34,7 @@ var cachedUserCount int64
 // previously fetched notices
 var cachedNotices model.ProductNotices
 
-func noticeMatchesConditions(conf *model.Config, client model.NoticeClientType, clientVersion, locale string, postCount, userCount int64, isSystemAdmin, isTeamAdmin bool, isCloud bool, sku string, notice *model.ProductNotice) (bool, error) {
+func noticeMatchesConditions(app AppIface, userId string, client model.NoticeClientType, clientVersion, locale string, postCount, userCount int64, isSystemAdmin, isTeamAdmin bool, isCloud bool, sku string, notice *model.ProductNotice) (bool, error) {
```

Comment:
I would prefer to pass the conf here, and pass an `store.Store` parameter too. Passing the app you are passing too much logic, and binding this to the app layer without explicit need.